mesonbuild/backend/vs2010backend.py

```diff
@@ -216,10 +221,10 @@ def generate_genlist_for_target(self, genlist: T.Union[build.GeneratedList, buil
                 ET.SubElement(cbs, 'Outputs').text = ';'.join(outfiles)
                 ET.SubElement(cbs, 'AdditionalInputs').text = ';'.join(deps)
 
-    def generate_custom_generator_commands(self, target, parent_node):
-        generator_output_files = []
-        custom_target_include_dirs = []
-        custom_target_output_files = []
+    def generate_custom_generator_commands(self, target, parent_node: ET.Element) -> T.Tuple[T.List[str], T.List[str], T.List[str]]:
```

Comment:
This returns List[str], according to my own analysis. And it is the intermediate route for generate_genlist_for_target -> get_primary_source_lang. So indeed why did I say File?